src/main/scala/cromwell/binding/Task.scala

```diff
@@ -1,50 +1,29 @@
 package cromwell.binding
 
+import java.util.regex.Pattern
+
 import cromwell.binding.AstTools.{AstNodeName, EnhancedAstNode}
-import cromwell.binding.command.{Command, ParameterCommandPart}
+import cromwell.binding.command.{CommandPart, ParameterCommandPart, StringCommandPart}
 import cromwell.parser.BackendType
 import cromwell.parser.WdlParser._
 
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
 import scala.language.postfixOps
+import scala.util.Try
 
 object Task {
   def apply(ast: Ast, backendType: BackendType, wdlSyntaxErrorFormatter: WdlSyntaxErrorFormatter): Task = {
     val name = ast.getAttribute("name").asInstanceOf[Terminal].getSourceString
     val declarations = ast.findAsts(AstNodeName.Declaration).map(Declaration(_, "name", wdlSyntaxErrorFormatter))
-
-    /* Examine all inputs to the task (which currently is only command parameters e.g. ${File x})
-     * And ensure that any inputs that have the same name also have the exact same definition.
-     * For example having a command line of `./script ${File x} ${String x}` is conflicting.
-     */
-    val commandParameters = ast.findAsts(AstNodeName.CommandParameter)
-    val parameterNames = commandParameters.map { _.getAttribute("name").sourceString() }.toSet
-    parameterNames.foreach { name =>
-      val paramsWithSameName = commandParameters.filter {
-        _.getAttribute("name").sourceString == name
-      }
-      ensureCommandParameterAstsMatch(paramsWithSameName, ast, wdlSyntaxErrorFormatter)
-    }
-
     val commandAsts = ast.findAsts(AstNodeName.Command)
-    if (commandAsts.size != 1) throw new UnsupportedOperationException("Expecting only one Command AST")
-    val command = Command(commandAsts.head, wdlSyntaxErrorFormatter)
-    val outputs = ast.findAsts(AstNodeName.Output) map {TaskOutput(_, wdlSyntaxErrorFormatter)}
-    new Task(name, declarations, command, outputs, ast, backendType)
-  }
-
-  private def ensureCommandParameterAstsMatch(paramAsts: Seq[Ast], taskAst: Ast, wdlSyntaxErrorFormatter: WdlSyntaxErrorFormatter) = {
-    paramAsts.headOption.foreach { firstParamAst =>
-      val sentinel = ParameterCommandPart(firstParamAst, wdlSyntaxErrorFormatter)
-      paramAsts.foreach { paramAst =>
-        val parsed = ParameterCommandPart(paramAst, wdlSyntaxErrorFormatter)
-        if (parsed != sentinel)
-          throw new SyntaxError(wdlSyntaxErrorFormatter.parametersWithSameNameMustHaveSameDefinition(
-            taskAst.getAttribute("name").asInstanceOf[Terminal],
-            paramAst.getAttribute("name").asInstanceOf[Terminal],
-            paramAsts.head.getAttribute("name").asInstanceOf[Terminal]
-          ))
-      }
+    if (commandAsts.size != 1) throw new UnsupportedOperationException("Expecting exactly one Command section")
```

Comment:
If `commandAsts` is a `List` this might be cleaner as a pattern match (one case for size is 1, one for not) via cons, but if it's `Seq` or something the cure is likely worse than the disease